node/test/go/dev-mode/test/lib/create-core-resources.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+const path = require('path');
+const fsp = require('fs').promises;
+const { Lambda } = require('@aws-sdk/client-lambda');
+const { IAM } = require('@aws-sdk/client-iam');
+const { STS } = require('@aws-sdk/client-sts');
+const log = require('log').get('test');
+const awsRequest = require('../utils/aws-request');
+const basename = require('./basename');
+
+const createLayer = async ({ layerName, filename }) => {
+  log.info('Publishing layer (%s) to AWS', filename);
+  const arn = (
+    await awsRequest(Lambda, 'publishLayerVersion', {
+      LayerName: layerName,
+      Content: { ZipFile: await fsp.readFile(filename) },
+    })
+  ).LayerVersionArn;
+  log.info('Layer ready %s', arn);
+  return arn;
+};
+
+const createLayers = async (config, layerTypes) => {
+  return Promise.all(
+    Array.from(layerTypes, async (layerType) => {
+      switch (layerType) {
+        case 'nodeInternal':
+          if (process.env.TEST_INTERNAL_LAYER_FILENAME) {
+            config.layerInternalArn = await createLayer({
+              layerName: `${basename}-internal`,
+              filename: path.resolve(process.env.TEST_INTERNAL_LAYER_FILENAME),
+              skipBuild: true,
+            });
+            return;
+          }
+          config.layerInternalArn = await createLayer({
+            layerName: `${basename}-internal`,
+            filename: path.resolve(__dirname, '../../dist/extension.internal.zip'),
+          });
+          return;
```

Comment:
Ahh I couldn't find this actually I was just using the `TEST_INTERNAL_LAYER_FILENAME` to run the tests 👍 I will change the path I was using in that environment variable to the default here.

Reply:
`TEST_INTERNAL_LAYER_FILENAME` is for testing custom version if _internal_ layer, I'd keep it. Also if that env var is set, there's no default internal layer loaded.

So it means you tested just _external_ extension with no internal loaded (How it could pass?)

Reply:
Lol this is a good question but it was running the tests alright before 🤷 Either way I updated this so it should be less confusing now

Reply:
This script seem to create external layer and create internal layer only if `TEST_INTERNAL_LAYER_FILENAME` var is set.

I think we should rather follow similar logic as in otel extension: https://github.com/serverless/console/blob/ce3e992f6322458519ec9b53e4c1ba25b3f29778/node/packages/aws-lambda-otel-extension/test/lib/create-core-resources.js#L49-L76

Reply:
Ahh okay got it 👍 I updated all of these names to be `EXTERNAL` instead of `INTERNAL`